src/routes.rs

```diff
@@ -83,9 +92,56 @@ pub async fn root_html() -> Result<Response, Rejection> {
     <html>
         <head>
             <title>Root</title>
+            <meta name="viewport" content="width=device-width, initial-scale=1">
+        </head>
+        <body>
+            <h1>Session Open Group Server</h1>
+            <p>This is a Session open group server.</p>
+        </body>
+    </html>
+    "#;
+    return Ok(warp::reply::html(body).into_response());
+}
+
+pub async fn fallback_html(room: String, query_map: HashMap<String, String>) -> Result<Response, Rejection> {
+    let public_key = match query_map.get("public_key") {
```

Comment:
You can use `unwrap_or("")` or `unwrap_or_default()` instaead of `match` here

Reply:
Sorry, due to my ineptitude with Rust I can't do it :(  
I get this error: `^^ expected struct `std::string::String`, found `str`` when using unwrap_or(""), and `method cannot be called on `std::option::Option<&std::string::String>` due to unsatisfied trait bounds` with the other.  
I've looked into the docs but I'm too smoothbrained to fix this. Feel free to edit it, it's probably something simple I'm missing

Reply:
I see, it's probably not worth trying to fix `String` `&str` incompatibility here. I think there is a better solution still if you don't need to actually use the value:
```rust
let contains_public_key = query_map.contains_key("public_key");
if !contains_public_key || room == "" {
...
}
```

Reply:
Okay, it works as intended.  
I have the push ready, I'm just waiting for Niels' comment in the conversation below to see if it's actually useful :P
Thanks for the heads-up 👍🏻